src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -0,0 +1,233 @@
+    return;
+  }
+
+  // Will store the solution of the ode
+  std::array<DataVector, 1> quaternion_to_integrate;
+  for (size_t i = stored_quaternions_and_times_.size();
+       i < omega_deriv_info.size(); i++) {
+    // Final time, initial time, and quaternion
+    const double t = omega_deriv_info[i].time;
+    const double t0 = omega_deriv_info[i - 1].time;
+    quaternion_to_integrate =
+        stored_quaternions_and_times_[i - 1].stored_quantities;
+
+    solve_quaternion_ode(&quaternion_to_integrate, t, t0);
+
+    // normalize quaternion
+    boost::math::quaternion<double> quat =
+        datavector_to_quaternion(quaternion_to_integrate[0]);
+    normalize_quaternion(make_not_null(&quat));
+
+    stored_quaternions_and_times_.emplace_back(
+        //  FunctionOfTimeHelpers::StoredInfo<1, false>{
+        t, std::array<DataVector, 1>{quaternion_to_datavector(quat)});
+  }
+
+  ASSERT(omega_deriv_info.size() == stored_quaternions_and_times_.size(),
+         "The number of stored omegas must be the same as the number of stored "
+         "quaternions after updating the missing quaternions. Now there are "
+             << omega_deriv_info.size() << " stored omegas and "
+             << stored_quaternions_and_times_.size() << " stored quaternions.");
+}
+
+template <size_t MaxDeriv>
+void QuaternionFunctionOfTime<MaxDeriv>::solve_quaternion_ode(
+    const gsl::not_null<std::array<DataVector, 1>*> quaternion_to_integrate,
+    const double t, const double t0) const noexcept {
+  // lambda that stores the internals of the ode
+  const auto quaternion_ode_system =
+      [this](const std::array<DataVector, 1>& state,
+             std::array<DataVector, 1>& dt_state, const double time) noexcept {
+        const std::array<DataVector, 1> omega = omega_f_of_t_.func(time);
+        dt_state[0] = DataVector{
+            {-0.5 * (state[0][1] * omega[0][0] + state[0][2] * omega[0][1] +
+                     state[0][3] * omega[0][2]),
+             0.5 * (state[0][0] * omega[0][0] + state[0][2] * omega[0][2] -
+                    state[0][3] * omega[0][1]),
+             0.5 * (state[0][0] * omega[0][1] + state[0][3] * omega[0][0] -
+                    state[0][1] * omega[0][2]),
+             0.5 * (state[0][0] * omega[0][2] + state[0][1] * omega[0][1] -
+                    state[0][2] * omega[0][0])}};
+      };
+
+  // Dense stepper
+  boost::numeric::odeint::dense_output_runge_kutta<
+      boost::numeric::odeint::controlled_runge_kutta<
+          boost::numeric::odeint::runge_kutta_dopri5<
+              std::array<DataVector, 1>>>>
+      dense_stepper = boost::numeric::odeint::make_dense_output(
+          1.0e-15, 1.0e-14,
+          boost::numeric::odeint::runge_kutta_dopri5<
+              std::array<DataVector, 1>>{});
+
+  // Initialize the stepper with initial quaternion (which is the quaternion
+  // passed in), initial time, and initial time step = 1e-3
+  dense_stepper.initialize(*quaternion_to_integrate, t0, 1e-3);
```

Comment:
How much does the initial time step matter?  Should it scale with something?

Reply:
The odeint uses adaptive time stepping to stay within error tolerances. Through trial and error, I've found that I get the same result for all time steps less than 1e-3 so I just chose that as a starting value.

Reply:
This number works for omega constant, linear, and quadratic in the tests. In practice the orbital angular velocity will be almost constant for most of inspiral. Just before merger it *may* be an issue depending on how small the simulation timesteps are. Since it's adaptive, would just setting it to a small value (say 1e-10) suffice, and then the integrator can change the time step however it likes to achieve the appropriate error? Or would that slow things down? (I also changed the current value to 1e-4 in the most recent fixup because changing to integrating quaternions rather than DataVectors somehow changed the error and tests were failing with 1e-3)